src/main/java/io/jenkins/infra/repository_permissions_updater/hosting/Hoster.java

```diff
@@ -210,17 +237,23 @@ boolean forkGitHub(String owner, String repo, String newName, List<String> maint
                     throw e;
             }
             if (newName != null) {
+                boolean renameResult = false;
                 for (int i = 0; i < 5; i++) {
                     try {
-                        r.renameTo(newName);
+                        renameResult = renameRepository(r, newName);
                         break;
                     } catch (HttpException e) {
+                        LOGGER.warn("Failed to rename repository from " + repo + " to " +  newName, e);
                         if (e.getResponseCode() == 422) {
-                            Thread.sleep(1000);
-                            renameRepository(r, newName);
+                            Thread.sleep(2000);
```

Comment:
I have the impression that github somehow claims the repo is forked and available but when you try to rename the fork operation is still in progress. So give the rename more time.